bedms/train.py

```diff
@@ -0,0 +1,242 @@
+""" This is the training script with which the user can train their own models."""
+
+import logging
+import torch
+from torch import nn
+from torch import optim
+from sklearn.metrics import (
+    precision_score,
+    recall_score,
+    f1_score,
+)
+import yaml
+from .utils_train import (
+    load_from_dir,
+    accumulate_data,
+    training_encoding,
+    data_loader,
+    train_model,
+    plot_learning_curve,
+    model_testing,
+    plot_confusion_matrix,
+    auc_roc_curve,
+)
+from .const import PROJECT_NAME
+from .model import BoWSTModel
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(PROJECT_NAME)
+
+
+class AttrStandardizerTrainer:
+    """
+    This is the training class responsible for
+    managing the training process for the standardizer model.
+    """
+
+    def __init__(self, config: str) -> None:
+        """
+        Initializes the TrainStandardizer object with the given configuration.
+
+        :param str config: Path to the config file which has the training parameters provided by the user.
+        """
+        self.label_encoder = None
+        self.vectorizer = None
+        self.train_loader = None
+        self.val_loader = None
+        self.test_loader = None
+        self.output_size = None
+        self.criterion = None
+        self.train_accuracies = None
+        self.val_accuracies = None
+        self.train_losses = None
+        self.val_losses = None
+        self.model = None
+        self.fpr = None
+        self.tpr = None
+        self.roc_auc = None
+        self.all_labels = None
+        self.all_preds = None
+
+        with open(config, "r") as file:
+            self.config = yaml.safe_load(file)
+
+    def load_encode_data(self) -> None:
+        """
+        Loads and prepares the encoded training, testing and validation datasets.
+        """
+        values_files_list = load_from_dir(self.config["dataset"]["values_dir_pth"])
+        headers_files_list = load_from_dir(self.config["dataset"]["headers_dir_pth"])
+
+        if len(values_files_list) != len(headers_files_list):
+            logger.error(
+                f"Mismatch in number of value files ({len(values_files_list)}) \
+                and header files ({len(headers_files_list)})"
+            )
+            return
+
+        total_files = len(values_files_list)
+
+        paired_files = list(zip(values_files_list, headers_files_list))
+
+        train_size = self.config["data_split"]["train_set"]
+        test_size = self.config["data_split"]["test_set"]
+        val_size = self.config["data_split"]["val_set"]
```

Comment:
I think that these should be percentages, not counts of files.. like i want to specify `0.7`, `0.2`, and `0.1`